app/models/BusinessEntity.scala

```diff
@@ -47,15 +47,15 @@ sealed trait BusinessEntity {
 
 object BusinessEntity {
   val reads: Reads[BusinessEntity] = Reads { json =>
-    Json.fromJson(json)(IncorporatedEntity.format).orElse(Json.fromJson(json)(SoleTrader.format)).orElse(Json.fromJson(json)(GeneralPartnership.format))
+    Json.fromJson(json)(IncorporatedEntity.format).orElse(Json.fromJson(json)(SoleTrader.format)).orElse(Json.fromJson(json)(PartnershipIdEntity.format))
   }
   val writes: Writes[BusinessEntity] = Writes {
     case incorporatedEntity: IncorporatedEntity =>
       Json.toJson(incorporatedEntity)(IncorporatedEntity.format)
     case soleTrader@SoleTrader(_, _, _, _, _, _, _, _, _) =>
       Json.toJson(soleTrader)(SoleTrader.format)
-    case generalPartnership@GeneralPartnership(_, _, _, _, _, _) =>
-      Json.toJson(generalPartnership)(GeneralPartnership.format)
+    case generalPartnership@PartnershipIdEntity(_, _, _, _, _, _, _) =>
```

Comment:
Given that this case will cover anything returned from partnership ID now we should probably rename the value to something like "partnershipEntity", similar to the incorp ID case